src/apps/solarflare/solarflare.lua

```diff
@@ -0,0 +1,332 @@
+module(...,package.seeall)
+
+local lib      = require("core.lib")
+local freelist = require("core.freelist")
+local packet   = require("core.packet")
+                 require("apps.solarflare.ef_vi_h")
+local pci      = require("lib.hardware.pci")
+local ethernet = require("lib.protocol.ethernet")
+
+local ffi = require("ffi")
+local C = ffi.C
+
+local RECEIVE_BUFFER_COUNT = 256
+local FLUSH_RECEIVE_QUEUE_THRESHOLD = 32
+local TX_BUFFER_COUNT = 256
+
+local ciul = ffi.load("ciul")
+
+local ef_vi_version = ffi.string(ciul.ef_vi_version_str())
+local required_ef_vi_version = "201502"
+
+if ef_vi_version ~= required_ef_vi_version then
+   error(string.format("ef_vi library does not have the correct version identified, need %s, got %s",
+                       required_ef_vi_version, ef_vi_version))
+end
+
+-- common utility functions
+
+ffi.cdef[[
+char *strerror(int errnum);
+]]
+
+local function try (rc, message)
+   if rc < 0 then
+      error(string.format("%s failed: %s", message,
+                          ffi.string(C.strerror(ffi.errno()))))
+   end
+   return rc
+end
+
+SolarFlareNic = {}
+SolarFlareNic.__index = SolarFlareNic
+SolarFlareNic.version = ef_vi_version
+
+driver = SolarFlareNic
```

Comment:
I think the `driver = ...` lines in `solarflare.lua` and `intel10g.lua` could use a brief comment to say that these are to make it easier for driver modules to be used interchangably.